bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/QuantityTypeArithmeticGroupFunction.java

```diff
@@ -218,12 +214,11 @@ public State calculate(@Nullable Set<Item> items) {
                 if (!isCompatible(item)) {
                     continue;
                 }
-                QuantityType itemState = item.getStateAs(QuantityType.class);
+                QuantityType itemState = normalizedQuantityType(item.getStateAs(QuantityType.class));
                 if (itemState == null) {
                     continue;
                 }
-                itemState = itemState.toInvertibleUnit(targetUnit);
-                if (itemState != null && (min == null || min.compareTo(itemState) > 0)) {
+                if (min == null || min.compareTo(itemState) > 0) {
```

Comment:
These comparison operators are the wrong way around. But that is because the compareTo method is wrong in the first place. See https://github.com/openhab/openhab-core/pull/4571